libs/blocks/preflight/panels/general.js

```diff
@@ -23,44 +22,42 @@ async function getStatus(suppliedPath) {
   return { url, edit, preview, live };
 }
 
-function findLinks(parent, selector) {
-  return [...parent.querySelectorAll(selector)].map((el) => {
-    const { modalPath, modalHash, path: fragmentPath } = el.dataset;
-    let dataPath = modalPath ? `${modalPath}${modalHash}` : fragmentPath;
-    // Find out if this is a fully qualified URL
-    try {
-      const url = new URL(dataPath);
-      dataPath = url.pathname;
-    } catch (e) {
-      // not a fully qualified URL
-    }
-    let path = dataPath ? `${window.location.origin}${dataPath}` : el.href;
-    path = path.endsWith('/') ? `${path}index` : path;
-    return getStatus(path);
+async function getStatuses() {
+  Object.keys(content.value).forEach((key) => {
```

Comment:
I wonder whether this is a good use case for `for...of`, since `forEach` might lead to unexpected results within `async/await` - https://gist.github.com/joeytwiddle/37d2085425c049629b80956d3c618971

Reply:
@overmyheadandbody normally, I would [agree](https://github.com/adobecom/milo/blob/main/libs/utils/utils.js#L671).

If we were concerned about doing something in a specific order, or if we do not want the next item to be fetched until the current is done, a for loop would be the best use. The other use case is if we are trying to set values outside of the loop.

In this case, we are using forEach's poor async/await support as an advantage. All the requests are sent nearly concurrently, and they come back when they come back (and out of order). We then set the content within the context of the item in the forEach. This has fairly substantial performance benefits since all requests are sent at nearly the same time.

Based on some quick calculations, i see:
1. ForEach - 1917ms
2. For loop - 20895ms

I'm inclined to keep this as is unless we:

a) find a race condition with setting the content inside the loop
b) find out helix admin cannot handle so many concurrent requests

Using for loop...
<img width="111" alt="Screen Shot 2023-03-07 at 10 40 45 AM" src="https://user-images.githubusercontent.com/1972095/223505197-41d2daa3-4135-4c89-a4bf-2dd6991717b6.png">

Using forEach...
<img width="197" alt="Screen Shot 2023-03-07 at 10 43 51 AM" src="https://user-images.githubusercontent.com/1972095/223505651-e60a12ec-ccdc-4f81-b488-b7066d180776.png">

Reply:
I agree that running in parallel is preferred here.  I think you should include a comment explaining this for future devs though.

Do not change this - but this is a case where the Promise.then(...) form is clearer to me.